pkg/minikube/registry/drvs/docker/docker.go

```diff
@@ -114,7 +114,49 @@ func status() registry.State {
 	return registry.State{Error: err, Installed: true, Healthy: false, Doc: docURL}
 }
 
-//suggestFix matches a stderr with possible fix for the docker driver
+// checkNeedsImprovement if overlay mod is installed on a system
+func checkNeedsImprovement() registry.State {
+	if runtime.GOOS == "linux" {
+		return checkOverlayMod()
+	} // TODO #8540: on non-linux check if docker desktop has enough CPU/memory
+	return registry.State{Installed: true, Healthy: true}
+}
+
+// checkOverlayMod checks if
+func checkOverlayMod() registry.State {
+	ctx, cancel := context.WithTimeout(context.Background(), 6*time.Second)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, "modprobe", "overlay")
+	_, err := cmd.Output()
+	if err != nil {
+		// try a different way
+		cmd = exec.CommandContext(ctx, "uname", "-r")
+		out, err := cmd.Output()
+		if ctx.Err() == context.DeadlineExceeded {
+			glog.Warningf("%q timed out checking for ", strings.Join(cmd.Args, " "))
+			return registry.State{NeedsImprovement: true, Installed: true, Healthy: true, Fix: "enable overlayfs kernel module on your Linux"}
+		}
+		if err != nil {
+			glog.Warningf("couldn't verify the linux distro's uname : %s", err)
+			return registry.State{NeedsImprovement: true, Installed: true, Healthy: true, Fix: "enable overlayfs kernel module on your Linux"}
+		}
+		path := fmt.Sprintf("/lib/modules/%s/modules.builtin", string(out))
+		cmd = exec.CommandContext(ctx, "cat", path)
+		out, err = cmd.Output()
+		if err != nil {
+			glog.Warningf("overlay module was not found in %q", path)
+			return registry.State{NeedsImprovement: true, Installed: true, Healthy: true, Fix: "enable overlayfs kernel module on your Linux"}
+		}
+		if strings.Contains(string(out), "overlay") { // success
+			return registry.State{NeedsImprovement: false, Installed: true, Healthy: true}
+		}
+		glog.Warningf("overlay module was not found")
```

Comment:
This is only checking the builtin modules, not the actual kernel modules ?

e.g. /lib/modules/5.4.40/kernel/fs/overlayfs/overlay.ko

Normally we use `modprobe` when dealing with modules.

`modprobe -n overlay`

Reply:
> What are we trying to guard against here ? Users that are using `vfs` driver for "special" reasons (NTFS) ? #7933
> 
> Or to avoid destroying the docker installation of older docker users ? #7626 Or users that are using btrfs ? #6167

on our current roadmap and capacitites we dont have room for supporting non-overlay in preload
so technically I am warnning the users that till we solve this issue, they will have a bad time with minikibe if they dont have overlay enabled.

I have personally seen people who were using minikube with non-overlay and it was extremely slow and it kept crashing... sometimies it works but....very poor performance, so for that reason I decided to maybe add a message and let the users know...

I could reword the message and  till this issue is solved, they need to use it minikube with overlay otherwsie they will see performance problems
https://github.com/kubernetes/minikube/issues/7626



what do u think @afbjorklund

Reply:
> iltin modules, not the actual kernel modules ?
> 
> e.g. /lib/modules/5.4.40/kernel/fs/overlayfs/overlay.ko

this is used as the second chance trying, if the first method didn't work... I remember on chromebook or maybe another platofrom the first way didnt' work but the second way worked.

what do u think ? @afbjorklund

Reply:
Ah, didn't read enough of the context - it is fine as a fallback solution, doing it "manually"